integrations/eureka/eureka/src/main/java/io/helidon/integrations/eureka/EurekaRegistrationHttpFeature.java

```diff
@@ -0,0 +1,668 @@
+    private boolean cancel(Http1Client client, String appName, String id) {
+        try (var response = client
+             .delete("/v2/apps/" + Objects.requireNonNull(appName, "appName") + "/" + Objects.requireNonNull(id, "id"))
+             .request()) {
+            if (response.status().family() == SUCCESSFUL) {
+                if (LOGGER.isLoggable(DEBUG)) {
+                    LOGGER.log(DEBUG,
+                               "DELETE /v2/apps/" + appName + "/" + id + ": " + response.status());
+                }
+                return true;
+            } else if (LOGGER.isLoggable(WARNING)) {
+                LOGGER.log(WARNING,
+                           "DELETE /v2/apps/" + appName + "/" + id + ": " + response.status());
+            }
+        } catch (UncheckedIOException e) {
+            Throwable cause = e.getCause();
+            if (cause instanceof ConnectException) {
+                if (LOGGER.isLoggable(WARNING)) {
+                    LOGGER.log(WARNING,
+                               "Eureka Server ("
+                               + this.client.prototype().baseUri().orElse(null) // volatile read
+                               + ") not reachable",
+                               e);
+                }
+            } else {
+                throw e;
+            }
+        }
+        return false;
+    }
+
+    private void createAndStartRenewalLoop(JsonObject instanceInfo, Http1Client client) {
+        if (LOGGER.isLoggable(DEBUG)) {
+            LOGGER.log(DEBUG,
+                       "Creating and starting Eureka lease renewal loop");
+        }
+        long sleepTimeInMilliSeconds = instanceInfo.getJsonObject("instance") // https://github.com/Netflix/eureka/blob/v2.0.4/eureka-client/src/main/java/com/netflix/appinfo/InstanceInfo.java#L55
+            .getJsonObject("leaseInfo")
+            .getInt("renewalIntervalInSecs") * 1000L;
+        this.renewer = Thread.ofVirtual() // volatile write
+            .name("Eureka lease renewer")
+            .uncaughtExceptionHandler((t, e) -> {
+                    if (LOGGER.isLoggable(ERROR)) {
+                        LOGGER.log(ERROR, e.getMessage(), e);
+                    }
+                    this.stop = true; // volatile write
+                })
+            .start(() -> {
+                    // Simplest possible heartbeat loop; nothing more complicated is needed.
+                    while (!this.stop) { // volatile read
+                        JsonObject newInstanceInfo = this.renew();
+                        if (newInstanceInfo != this.instanceInfo) { // volatile read
+                            // The server gave us something new for some reason; use it.
+                            this.instanceInfo = newInstanceInfo; // volatile write
+                        }
+                        try {
+                            sleep(sleepTimeInMilliSeconds);
+                        } catch (InterruptedException e) {
+                        }
```

Comment:
Not even a log message for the muffled exception?

Reply:
Oversight; fixed in next commit